File: business_logic/shipping_service.cs
 namespace InternTask.business_logic;
public class ShippingService
{
    public void ShipItems(List<IShippable> items)
    {
        Console.WriteLine("** Shipment notice **");
        
        double totalWeight = 0;
        
        foreach (var group in items.GroupBy(i => i.GetName()))
        {
            var item = group.First();
            int count = group.Count();
            double itemWeight = item.GetWeight();
            totalWeight += count * itemWeight;
            
            Console.WriteLine($"{count}x {item.GetName()} {itemWeight * 1000}g");
        }
        
        Console.WriteLine($"Total package weight {totalWeight}kg");
    }
}

File: lib/ShippingService.java
class ShippingService {
    public void shipItems(java.util.List<Shippable> items) {
        System.out.println("** Shipment notice **");
        
        double totalWeight = 0;
        
        java.util.Map<String, java.util.List<Shippable>> itemGroups = new java.util.HashMap<>();
        for (Shippable item : items) {
            String name = item.getName();
            if (!itemGroups.containsKey(name)) {
                itemGroups.put(name, new java.util.ArrayList<>());
            }
            itemGroups.get(name).add(item);
            totalWeight += item.getWeight();
        }
        
        for (String name : itemGroups.keySet()) {
            java.util.List<Shippable> group = itemGroups.get(name);
            Shippable firstItem = group.get(0);
            int count = group.size();
            double itemWeight = firstItem.getWeight();
            
            System.out.println(count + "x " + name + " " + (itemWeight * 1000) + "g");
        }
        
        System.out.println("Total package weight " + totalWeight + "kg");
    }
}
